src/OpenTelemetry.Extensions/TracerProviderBuilderExtensions.cs

```diff
@@ -53,14 +54,42 @@ public static class TracerProviderBuilderExtensions
     /// Adds the <see cref="BaggageActivityProcessor"/> to the <see cref="TracerProviderBuilder"/>.
     /// </summary>
     /// <param name="builder"><see cref="TracerProviderBuilder"/> to add the <see cref="BaggageActivityProcessor"/> to.</param>
+    /// <param name="baggageKeyPredicate">Predicate to determine which baggage keys should be added to the activity.</param>
     /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
     public static TracerProviderBuilder AddBaggageActivityProcessor(
-        this TracerProviderBuilder builder)
+        this TracerProviderBuilder builder,
+        Predicate<string> baggageKeyPredicate)
     {
         Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNull(baggageKeyPredicate);
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
+        return builder.AddProcessor(new BaggageActivityProcessor(baggageKeyPredicate));
+#pragma warning restore CA2000 // Dispose objects before losing scope
+    }
+
+    /// <summary>
+    /// Adds the <see cref="BaggageActivityProcessor"/> to the <see cref="TracerProviderBuilder"/>.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> to add the <see cref="BaggageActivityProcessor"/> to.</param>
+    /// <param name="expr">Regular expression to determine which baggage keys should be added to the activity.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddBaggageActivityProcessor(
+        this TracerProviderBuilder builder, string expr)
+    {
+        Guard.ThrowIfNull(builder);
+        Guard.ThrowIfNullOrEmpty(expr);
+
+        if (expr == "*")
+        {
+#pragma warning disable CA2000 // Dispose objects before losing scope
```

Comment:
Just FYI if you use the factory pattern it _should_ make these CA2000 warnings go away:

```csharp
return builder.AddProcessor(sp => new BaggageActivityProcessor(_ => true));
```